fix(lint): landing_partner/user — eslint-config-next babel/eslint-parser not found (pre-existing)

Scheduler: needs-swe-sonnet-1

## 배경

landing_partner/user의 `npm run lint` 단계가 다음 오류로 실패:

```
Error: Cannot find module 'next/dist/compiled/babel/eslint-parser'
Require stack:
- /node_modules/eslint-config-next/dist/parser.js
- /node_modules/eslint-config-next/dist/index.js
- /node_modules/eslint-config-next/dist/core-web-vitals.js
```

이 오류는 2026-03-08 이전부터 CI에 존재 (dev 브랜치 CI run #22812244118 등).

## 영향 범위

- `apps/landing_partner/eslint.config.mjs` + `apps/landing_user/eslint.config.mjs`
- `lint-and-build-landing-partner`, `lint-and-build-landing-user` CI job
- Dependabot PR #1626, #1627이 ESLint 오류로 ci-result 실패

## 원인 분석

`eslint-config-next/core-web-vitals` 가 `next/dist/compiled/babel/eslint-parser`를 require하는데, npm workspace 환경에서 `next`가 `eslint-config-next`의 위치와 다른 `node_modules`에 설치될 때 모듈 해석 실패.

## 수정 가이드

1. `eslint.config.mjs`에서 `eslint-config-next/core-web-vitals` import 방식 변경
2. Next.js 16.x 권장 flat config 패턴으로 업데이트
3. `eslint-config-next`를 workspace devDep 대신 root devDep으로 이동 검토

## 우선순위

P3-low — 기존 기능에 영향 없음. landing 앱 CI 린트만 실패.

--- a/apps/landing_partner/eslint.config.mjs
+++ b/apps/landing_partner/eslint.config.mjs
@@ -2,10 +2,15 @@
 // eslint-config-next/core-web-vitals. The core-web-vitals entry point loads
 // eslint-config-next/dist/parser.js which requires
 // next/dist/compiled/babel/eslint-parser — a file removed in next@16+.
-// Direct plugin usage bypasses parser.js entirely.
+// Direct plugin usage bypasses parser.js entirely. React/react-hooks/jsx-a11y
+// plugins are wired explicitly to preserve the lint coverage that
+// eslint-config-next/core-web-vitals would have provided via its base config.
 import { defineConfig, globalIgnores } from "eslint/config";
 import nextPlugin from "@next/eslint-plugin-next";
 import nextTs from "eslint-config-next/typescript";
+import reactPlugin from "eslint-plugin-react";
+import reactHooksPlugin from "eslint-plugin-react-hooks";
+import jsxA11y from "eslint-plugin-jsx-a11y";
 
 const eslintConfig = defineConfig([
   {
@@ -17,15 +22,25 @@ const eslintConfig = defineConfig([
       ...nextPlugin.configs["core-web-vitals"].rules,
     },
   },
-  ...nextTs,
   // eslint-plugin-react@7.x uses context.getFilename() which is not available
   // in flat config. Providing version explicitly avoids the detectReactVersion
   // call that triggers the crash. Remove once eslint-plugin-react supports flat config.
   {
+    plugins: {
+      react: reactPlugin,
+      "react-hooks": reactHooksPlugin,
+      "jsx-a11y": jsxA11y,
+    },
+    rules: {
+      ...reactPlugin.configs["jsx-runtime"].rules,
+      ...reactHooksPlugin.configs.recommended.rules,
+      ...jsxA11y.configs.recommended.rules,
+    },
     settings: {
       react: { version: "19" },
     },
   },
+  ...nextTs,
   // Override default ignores of eslint-config-next.
   globalIgnores([
     // Default ignores of eslint-config-next:

--- a/apps/landing_partner/eslint.config.mjs
+++ b/apps/landing_partner/eslint.config.mjs
@@ -1,9 +1,22 @@
+// Fix #1721: Import @next/eslint-plugin-next directly instead of
+// eslint-config-next/core-web-vitals. The core-web-vitals entry point loads
+// eslint-config-next/dist/parser.js which requires
+// next/dist/compiled/babel/eslint-parser — a file removed in next@16+.
+// Direct plugin usage bypasses parser.js entirely.
 import { defineConfig, globalIgnores } from "eslint/config";
-import nextVitals from "eslint-config-next/core-web-vitals";
+import nextPlugin from "@next/eslint-plugin-next";
 import nextTs from "eslint-config-next/typescript";
 
 const eslintConfig = defineConfig([
-  ...nextVitals,
+  {
+    plugins: {
+      "@next/next": nextPlugin,
+    },
+    rules: {
+      ...nextPlugin.configs.recommended.rules,
+      ...nextPlugin.configs["core-web-vitals"].rules,
+    },
+  },
   ...nextTs,
   // eslint-plugin-react@7.x uses context.getFilename() which is not available
   // in flat config. Providing version explicitly avoids the detectReactVersion

--- a/apps/landing_partner/eslint.config.mjs
+++ b/apps/landing_partner/eslint.config.mjs
@@ -1,54 +1,73 @@
-// Fix #1721: Import @next/eslint-plugin-next directly instead of
-// eslint-config-next/core-web-vitals. The core-web-vitals entry point loads
-// eslint-config-next/dist/parser.js which requires
-// next/dist/compiled/babel/eslint-parser — a file removed in next@16+.
-// Direct plugin usage bypasses parser.js entirely. React/react-hooks/jsx-a11y
-// plugins are wired explicitly to preserve the lint coverage that
-// eslint-config-next/core-web-vitals would have provided via its base config.
+// Fix #1721: eslint-config-next/dist/index.js loads ./parser which requires
+// next/dist/compiled/babel/eslint-parser — a file removed in Next.js 16+.
+// We replicate the full eslint-config-next base config inline, replacing the
+// broken babel parser with @typescript-eslint/parser (already used by the
+// 'next/typescript' overlay for .ts/.tsx). All plugins, rules, globs, globals,
+// and import-resolver settings from eslint-config-next v16.2.3 are preserved.
 import { defineConfig, globalIgnores } from "eslint/config";
 import nextPlugin from "@next/eslint-plugin-next";
 import nextTs from "eslint-config-next/typescript";
 import reactPlugin from "eslint-plugin-react";
 import reactHooksPlugin from "eslint-plugin-react-hooks";
 import jsxA11y from "eslint-plugin-jsx-a11y";
+import * as importPlugin from "eslint-plugin-import";
+import globals from "globals";
 
 const eslintConfig = defineConfig([
+  // Mirrors eslint-config-next/dist/index.js "next" config block,
+  // with @typescript-eslint/parser replacing the missing babel parser.
   {
+    name: "next",
+    files: ["**/*.{js,jsx,mjs,ts,tsx,mts,cts}"],
     plugins: {
       "@next/next": nextPlugin,
-    },
-    rules: {
-      ...nextPlugin.configs.recommended.rules,
-      ...nextPlugin.configs["core-web-vitals"].rules,
-    },
-  },
-  // eslint-plugin-react@7.x uses context.getFilename() which is not available
-  // in flat config. Providing version explicitly avoids the detectReactVersion
-  // call that triggers the crash. Remove once eslint-plugin-react supports flat config.
-  {
-    plugins: {
       react: reactPlugin,
       "react-hooks": reactHooksPlugin,
       "jsx-a11y": jsxA11y,
+      import: importPlugin,
+    },
+    languageOptions: {
+      globals: {
+        ...globals.browser,
+        ...globals.node,
+      },
+    },
+    settings: {
+      // eslint-plugin-react@7.x uses context.getFilename() in flat config;
+      // pinning version avoids the detectReactVersion crash.
+      react: { version: "19" },
+      "import/parsers": {
+        "@typescript-eslint/parser": [".ts", ".mts", ".cts", ".tsx", ".d.ts"],
+      },
+      "import/resolver": {
+        node: { extensions: [".js", ".jsx", ".ts", ".tsx"] },
+        typescript: { alwaysTryTypes: true },
+      },
     },
     rules: {
       ...reactPlugin.configs["jsx-runtime"].rules,
       ...reactHooksPlugin.configs.recommended.rules,
-      ...jsxA11y.configs.recommended.rules,
-    },
-    settings: {
-      react: { version: "19" },
+      ...nextPlugin.configs.recommended.rules,
+      ...nextPlugin.configs["core-web-vitals"].rules,
+      "import/no-anonymous-default-export": "warn",
+      "jsx-a11y/alt-text": [
+        "warn",
+        { elements: ["img"], img: ["Image"] },
+      ],
+      "jsx-a11y/aria-props": "warn",
+      "jsx-a11y/aria-proptypes": "warn",
+      "jsx-a11y/aria-unsupported-elements": "warn",
+      "jsx-a11y/role-has-required-aria-props": "warn",
+      "jsx-a11y/role-supports-aria-props": "warn",
+      "react/jsx-no-target-blank": "off",
+      "react/no-unknown-property": "off",
+      "react/react-in-jsx-scope": "off",
+      "react/prop-types": "off",
     },
   },
+  // TypeScript overlay: @typescript-eslint/parser + recommended rules for .ts/.tsx
   ...nextTs,
-  // Override default ignores of eslint-config-next.
-  globalIgnores([
-    // Default ignores of eslint-config-next:
-    ".next/**",
-    "out/**",
-    "build/**",
-    "next-env.d.ts",
-  ]),
+  globalIgnores([".next/**", "out/**", "build/**", "next-env.d.ts"]),
 ]);
 
 export default eslintConfig;

--- a/apps/landing_user/eslint.config.mjs
+++ b/apps/landing_user/eslint.config.mjs
@@ -2,10 +2,15 @@
 // eslint-config-next/core-web-vitals. The core-web-vitals entry point loads
 // eslint-config-next/dist/parser.js which requires
 // next/dist/compiled/babel/eslint-parser — a file removed in next@16+.
-// Direct plugin usage bypasses parser.js entirely.
+// Direct plugin usage bypasses parser.js entirely. React/react-hooks/jsx-a11y
+// plugins are wired explicitly to preserve the lint coverage that
+// eslint-config-next/core-web-vitals would have provided via its base config.
 import { defineConfig, globalIgnores } from "eslint/config";
 import nextPlugin from "@next/eslint-plugin-next";
 import nextTs from "eslint-config-next/typescript";
+import reactPlugin from "eslint-plugin-react";
+import reactHooksPlugin from "eslint-plugin-react-hooks";
+import jsxA11y from "eslint-plugin-jsx-a11y";
 
 const eslintConfig = defineConfig([
   {
@@ -17,15 +22,25 @@ const eslintConfig = defineConfig([
       ...nextPlugin.configs["core-web-vitals"].rules,
     },
   },
-  ...nextTs,
   // eslint-plugin-react@7.x uses context.getFilename() which is not available
   // in flat config. Providing version explicitly avoids the detectReactVersion
   // call that triggers the crash. Remove once eslint-plugin-react supports flat config.
   {
+    plugins: {
+      react: reactPlugin,
+      "react-hooks": reactHooksPlugin,
+      "jsx-a11y": jsxA11y,
+    },
+    rules: {
+      ...reactPlugin.configs["jsx-runtime"].rules,
+      ...reactHooksPlugin.configs.recommended.rules,
+      ...jsxA11y.configs.recommended.rules,
+    },
     settings: {
       react: { version: "19" },
     },
   },
+  ...nextTs,
   // Override default ignores of eslint-config-next.
   globalIgnores([
     // Default ignores of eslint-config-next:

--- a/apps/landing_user/eslint.config.mjs
+++ b/apps/landing_user/eslint.config.mjs
@@ -1,9 +1,22 @@
+// Fix #1721: Import @next/eslint-plugin-next directly instead of
+// eslint-config-next/core-web-vitals. The core-web-vitals entry point loads
+// eslint-config-next/dist/parser.js which requires
+// next/dist/compiled/babel/eslint-parser — a file removed in next@16+.
+// Direct plugin usage bypasses parser.js entirely.
 import { defineConfig, globalIgnores } from "eslint/config";
-import nextVitals from "eslint-config-next/core-web-vitals";
+import nextPlugin from "@next/eslint-plugin-next";
 import nextTs from "eslint-config-next/typescript";
 
 const eslintConfig = defineConfig([
-  ...nextVitals,
+  {
+    plugins: {
+      "@next/next": nextPlugin,
+    },
+    rules: {
+      ...nextPlugin.configs.recommended.rules,
+      ...nextPlugin.configs["core-web-vitals"].rules,
+    },
+  },
   ...nextTs,
   // eslint-plugin-react@7.x uses context.getFilename() which is not available
   // in flat config. Providing version explicitly avoids the detectReactVersion

--- a/apps/landing_user/eslint.config.mjs
+++ b/apps/landing_user/eslint.config.mjs
@@ -1,54 +1,73 @@
-// Fix #1721: Import @next/eslint-plugin-next directly instead of
-// eslint-config-next/core-web-vitals. The core-web-vitals entry point loads
-// eslint-config-next/dist/parser.js which requires
-// next/dist/compiled/babel/eslint-parser — a file removed in next@16+.
-// Direct plugin usage bypasses parser.js entirely. React/react-hooks/jsx-a11y
-// plugins are wired explicitly to preserve the lint coverage that
-// eslint-config-next/core-web-vitals would have provided via its base config.
+// Fix #1721: eslint-config-next/dist/index.js loads ./parser which requires
+// next/dist/compiled/babel/eslint-parser — a file removed in Next.js 16+.
+// We replicate the full eslint-config-next base config inline, replacing the
+// broken babel parser with @typescript-eslint/parser (already used by the
+// 'next/typescript' overlay for .ts/.tsx). All plugins, rules, globs, globals,
+// and import-resolver settings from eslint-config-next v16.2.3 are preserved.
 import { defineConfig, globalIgnores } from "eslint/config";
 import nextPlugin from "@next/eslint-plugin-next";
 import nextTs from "eslint-config-next/typescript";
 import reactPlugin from "eslint-plugin-react";
 import reactHooksPlugin from "eslint-plugin-react-hooks";
 import jsxA11y from "eslint-plugin-jsx-a11y";
+import * as importPlugin from "eslint-plugin-import";
+import globals from "globals";
 
 const eslintConfig = defineConfig([
+  // Mirrors eslint-config-next/dist/index.js "next" config block,
+  // with @typescript-eslint/parser replacing the missing babel parser.
   {
+    name: "next",
+    files: ["**/*.{js,jsx,mjs,ts,tsx,mts,cts}"],
     plugins: {
       "@next/next": nextPlugin,
-    },
-    rules: {
-      ...nextPlugin.configs.recommended.rules,
-      ...nextPlugin.configs["core-web-vitals"].rules,
-    },
-  },
-  // eslint-plugin-react@7.x uses context.getFilename() which is not available
-  // in flat config. Providing version explicitly avoids the detectReactVersion
-  // call that triggers the crash. Remove once eslint-plugin-react supports flat config.
-  {
-    plugins: {
       react: reactPlugin,
       "react-hooks": reactHooksPlugin,
       "jsx-a11y": jsxA11y,
+      import: importPlugin,
+    },
+    languageOptions: {
+      globals: {
+        ...globals.browser,
+        ...globals.node,
+      },
+    },
+    settings: {
+      // eslint-plugin-react@7.x uses context.getFilename() in flat config;
+      // pinning version avoids the detectReactVersion crash.
+      react: { version: "19" },
+      "import/parsers": {
+        "@typescript-eslint/parser": [".ts", ".mts", ".cts", ".tsx", ".d.ts"],
+      },
+      "import/resolver": {
+        node: { extensions: [".js", ".jsx", ".ts", ".tsx"] },
+        typescript: { alwaysTryTypes: true },
+      },
     },
     rules: {
       ...reactPlugin.configs["jsx-runtime"].rules,
       ...reactHooksPlugin.configs.recommended.rules,
-      ...jsxA11y.configs.recommended.rules,
-    },
-    settings: {
-      react: { version: "19" },
+      ...nextPlugin.configs.recommended.rules,
+      ...nextPlugin.configs["core-web-vitals"].rules,
+      "import/no-anonymous-default-export": "warn",
+      "jsx-a11y/alt-text": [
+        "warn",
+        { elements: ["img"], img: ["Image"] },
+      ],
+      "jsx-a11y/aria-props": "warn",
+      "jsx-a11y/aria-proptypes": "warn",
+      "jsx-a11y/aria-unsupported-elements": "warn",
+      "jsx-a11y/role-has-required-aria-props": "warn",
+      "jsx-a11y/role-supports-aria-props": "warn",
+      "react/jsx-no-target-blank": "off",
+      "react/no-unknown-property": "off",
+      "react/react-in-jsx-scope": "off",
+      "react/prop-types": "off",
     },
   },
+  // TypeScript overlay: @typescript-eslint/parser + recommended rules for .ts/.tsx
   ...nextTs,
-  // Override default ignores of eslint-config-next.
-  globalIgnores([
-    // Default ignores of eslint-config-next:
-    ".next/**",
-    "out/**",
-    "build/**",
-    "next-env.d.ts",
-  ]),
+  globalIgnores([".next/**", "out/**", "build/**", "next-env.d.ts"]),
 ]);
 
 export default eslintConfig;

